### PR TITLE
fixed all tests + some feature updates

### PR DESCRIPTION
**Pull request by agupta01**
Feature updates:
- added a force_start parameter to .fit() to allow user to lock a start date and bypass auto detection
- added a threshold to find_end_baseline
- order of operations in .fit() now first finds end date, then does SARIMAX, so that it needs to forecast fewer dates (possibly helps with speed but also makes calc_resid) simpler
   - this also means calc_resid now takes in a trimmed forecast series and doesn't need to trim on its own
- all util functions now are consistent with bounds, exclusive on start date, inclusive on end date

### CODE REVIEW

**Comment by agupta01** [comment]
hi please squash commits when you merge this

File: sigmet/au3_functions.py [hunk 1]
```diff
@@ -78,37 +78,48 @@ def find_start(series, user_start, user_end, ma_window=6):
...
     # filter series to window between start and end date, smooth moving average
     # NOTE: we inclusively filter with user_start b/c when we run the differencing it will no longer be part of the series
-    filtered_series = series.rolling(ma_window).mean().loc[(series.index >= user_start) & (series.index < user_end)]
+    filtered_series = series.rolling(ma_window).mean().loc[(
+        series.index >= user_start) & (series.index < user_end)]
     if filtered_series.hasnans:
-        raise ValueError("Moving average value too large for search window.")
+        raise ValueError("Moving average value too large for search window. Decrease the moving_average value or increase the size of the search window.")
 
     # special case if monotonic decreasing, want to check for largest first derivative
```
Comment (camille-004) [inline sigmet/au3_functions.py:86]: By largest you mean just magnitude right?

File: sigmet/au3_functions.py [hunk 2]
```diff
@@ -141,17 +152,17 @@ def ARIMA_predictor(series, start_date, params=(5, 1, 1)):
...
         model_fit = model.fit(disp=0)
 
         # Return the forecast as a pd.Series object
         return pd.Series(model_fit.forecast(steps)[0])
     except ValueError:
-        raise ValueError("Cannot provide an SARIMAX forecast for given trend")
+        raise ValueError("Cannot provide an ARIMA forecast for given trend")
 
 
-def SARIMAX_predictor(series, start_date, params=(5, 1, 1)):
```
Comment (camille-004) [inline sigmet/au3_functions.py:154]: Will this be ARIMA or SARIMAX?

**Review by camille-004 — APPROVED** [review]

File: sigmet/au3_functions.py (hunk 1 above)
Comment (agupta01) [reply sigmet/au3_functions.py:86]: yep

File: sigmet/au3_functions.py (hunk 2 above)
Comment (agupta01) [reply sigmet/au3_functions.py:154]: SARIMAX for now, but it'll be configurable once we build that functionality

**Pull request merged**